torchio/data/subject.py

```diff
@@ -214,19 +214,46 @@ def apply_inverse_transform(self, **kwargs) -> 'Subject':
 
     def clear_history(self) -> None:
         self.applied_transforms = []
-
-    def check_consistent_attribute(self, attribute: str) -> None:
-        values_dict = {}
+        
+    def check_consistent_attribute(self,
+                               attribute: str,
+                               rtol=1e-6,
+                               atol=1e-6,
+                               message: Optional[str] = None) -> None:
         iterable = self.get_images_dict(intensity_only=False).items()
-        for image_name, image in iterable:
-            values_dict[image_name] = getattr(image, attribute)
-        num_unique_values = len(set(values_dict.values()))
-        if num_unique_values > 1:
+        if message is None:
```

Comment:
I like this message. Why not just hard-coding it? I'm not sure the user needs the option to have a custom message. They can always catch the exception and print something else themselves.

Reply:
I didn't think about the user, rather when this method will be used for other checks so that the current methods these checks run on, can be reused and don't introduce breaking changes :)